tests/api_tests/nodes/test_views.py

```diff
@@ -863,6 +865,512 @@ def test_filtering_on_invalid_field(self):
+
+    def test_remove_contributor_non_admin_is_forbidden(self):
+        self.project.add_contributor(self.user_three, permissions=[permissions.READ, permissions.WRITE], visible=True, save=True)
+
+        res = self.app.delete(self.url_user_three, auth=self.user_two.auth, expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+        self.project.reload()
+        assert_in(self.user_three, self.project.contributors)
+
+    def test_remove_self_non_admin(self):
+        self.project.add_contributor(self.user_three, permissions=[permissions.READ, permissions.WRITE], visible=True, save=True)
+
+        res = self.app.delete(self.url_user_three, auth=self.user_three.auth)
+        self.project.reload()
+        assert_not_in(self.user_three, self.project.contributors)
+
+    def test_remove_contributor_non_contributor(self):
+        res = self.app.delete(self.url_user_two, auth=self.user_three.auth, expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+        self.project.reload()
+        assert_in(self.user_two, self.project.contributors)
+
+    def test_remove_contributor_not_logged_in(self):
+        res = self.app.delete(self.url_user_two, expect_errors=True)
+        assert_equal(res.status_code, 401)
+
+        self.project.reload()
+        assert_in(self.user_two, self.project.contributors)
+
+    def test_remove_non_contributor_admin(self):
+        res = self.app.delete(self.url_user_three, auth=self.user.auth, expect_errors=True)
+        assert_equal(res.status_code, 404)
+
+    def test_remove_non_existing_user_admin(self):
+        url_user_fake = '/{}nodes/{}/contributors/{}/'.format(API_BASE, self.project._id, 'fake')
+        res = self.app.delete(url_user_fake, auth=self.user.auth, expect_errors=True)
+        assert_equal(res.status_code, 404)
+
+    def test_remove_self_contributor_not_unique_admin(self):
+        self.project.add_permission(self.user_two, permissions.ADMIN, save=True)
+        res = self.app.delete(self.url_user, auth=self.user.auth)
+        assert_equal(res.status_code, 204)
+
+        self.project.reload()
+        assert_not_in(self.user, self.project.contributors)
+
+    def test_can_remove_self_as_contributor_not_unique_admin(self):
+        self.project.add_permission(self.user_two, permissions.ADMIN, save=True)
+        res = self.app.delete(self.url_user_two, auth=self.user_two.auth)
+        assert_equal(res.status_code, 204)
+
+        self.project.reload()
+        assert_not_in(self.user_two, self.project.contributors)
+
+    def test_remove_self_contributor_unique_admin(self):
+        res = self.app.delete(self.url_user, auth=self.user.auth, expect_errors=True)
+        assert_equal(res.status_code, 400)
+
+        self.project.reload()
+        assert_in(self.user, self.project.contributors)
+
+    def test_remove_only_bibliographic_contributor(self):
+        self.project.set_visible(self.user_two, False)
```

Comment:
Specifically: I think that the reason this test is failing is because self.user is the only admin (whether or not there are bibliographic contributors may be irrelevant).

If there is a rule that all projects must have at least 1 bibliographic contributor (?), then it should be tested separate from the rule that there must be at least 1 admin.